src/base/settingsstorage.h

```diff
@@ -62,4 +62,34 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
+    explicit StorableValue(const std::string &keyName, const QVariant &defaultValue = QVariant())
```

Comment:
Aren't you relying on the same `T()` when you pass an empty `QVariant` to `QVariant::value()`?